app/lib/navbar_visibility.rb

```diff
@@ -0,0 +1,15 @@
+module NavbarVisibility
+  NAVBAR_HIDEABLE = {
+    'announcements' => ['new', 'create'],
+    'asks' => ['new', 'create'],
+    'community_ressources' => ['new', 'show'],
+    'offers' => ['new', 'create'],
+    'public_pages' => ['about', 'contributions', 'landing_page', 'version'],
+    'thank_you' => ['show']
+  }
+
+  def self.shown_navbar?(controller, action, display_navbar)
+    always_display = NAVBAR_HIDEABLE[controller].nil? || !NAVBAR_HIDEABLE[controller].include?(action)
+    always_display || display_navbar
+  end
```

Comment:
One minor refactoring i might suggest:
```suggestion
  def self.shown?(controller, action, display_navbar)
    display_navbar || !hidden?(controller, action)
  end

  def self.hidden?(controller, action)
    NAVBAR_HIDEABLE[controller]&.include?(action)
  end 
```
* Since we're now using the method with the fully qualified module name, i think we don't need `navbar` in the method name
* Starting the conditional with `display_navbar ||` helps to clarify that the system system takes precedence over controller/action specific checks
* IMO, extracting the small `hidden?` method helps to simplify and make obvious the overall logic.

Reply:
You are right ! Done.